src/lib.rs

```diff
@@ -1,14 +1,259 @@
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
+#![no_std]
```

Comment:
Does using `no_std` like this mean we don't have to include the `clippy::alloc_instead_of_core` lint from https://github.com/rustls/rustls/pull/1378?

Reply:
I'm not super familiar with `no_std` setups, but given that so far in this crate nothing really needs `std`, it seemed most straightforward to unconditionally opt-in to `no_std` and make the `alloc` dependency conditional. Given the CI setup I think the lint is superfluous, because `--no-default-features` should trigger the same failures.